docs/targets/polkadot.rst

```diff
@@ -62,3 +61,45 @@ The following example shows how call flags can be used:
 .. include:: ../examples/polkadot/call_flags.sol
   :code: solidity
 
+
+Reverts and error data decoding
+_______________________________
+
+When a contract reverts, the returned error data is what
+`EVM would return <https://docs.soliditylang.org/en/v0.8.20/control-structures.html#panic-via-assert-and-error-via-require>`_.
+``assert()``, ``require()``, or ``revert()`` will revert the contract execution, where the revert reason 
+is supplied as the contracts output (if any). Solidity contracts can also revert with a `Panic` 
+(please refer to the 
+`Solidity language documentation <https://docs.soliditylang.org/en/v0.8.20/control-structures.html#panic-via-assert-and-error-via-require>`_.
+for more information about when `Panic` might be returned).
+Uncaught exceptions from calling and instantiating contracts or transferring funds will be bubbled 
+up back to the caller.
+
+The metadata contains all error variants that the contract knows about in the ``lang_error`` field.
+The 4 bytes `selector` of the error data can be seen as the enum discriminator. However, because 
+SCALE encoding does not allow discriminators larger than 1 byte, the hex-encoded error selector 
+is provided as the enum variant name in the metadata (the selector could also be calculated by
+reconstructing and hashing the error signature based on the enum variant types).
+
+.. note::
+
+    Solidity knows about `Error`, `Panic` and 
+    `custom errors <https://docs.soliditylang.org/en/latest/abi-spec.html#errors>`_.
```

Comment:
Please don't refer to solc/Ethereum Solidity as "Solidity". That suggests we are not Solidity

Reply:
> Please don't refer to solc/Ethereum Solidity as "Solidity"

I disagree. Solidity is what the language authors think it is. And that happens to be whats on soliditylang.org. We can't say we are a Solidity compiler when we in fact compile something that is not Solidity.

If you mean `solc/Ethereum` what is broadly understood as the _Solidity Language_, it doesn't follow to not call it _Solidity_. Saying  "Solidity knows about `X`" is correct.

Reply:
Also, I've called it "Ethereum Solidity" throughout the documentation. If you're going to change this, then change all the links that say "Ethereum Solidity".

Reply:
In one hand, Solidity for Solana has its own syntax, so we can see it as a different flavor. Calling Ethereum Solidity to refer to the original Solidity implementation is reasonable, given that we are slightly different than the original.

Reply:
> In one hand, Solidity for Solana has its own syntax, so we can see it as a different flavor. Calling Ethereum Solidity to refer to the original Solidity implementation is reasonable, given that we are slightly different than the original.

Exactly. On Polkadot there are also many small differences with Ethereum Solidity (or should be called Solidity on Ethereum/ Solidity on EVM)?

Reply:
> Your link suggests that https://docs.soliditylang.org/en/latest/abi-spec.html#errors is the authoritative answer.

Yes, intentionally so. I'd be largely surprised if this wasn't the authoritative answer. For a vast majority of anyone referring to the Solidity Language in any way.

> Where does that leave Solang?

This is concisely explained in literally the next sentence:

    Solang does not yet support custom errors.

> Also, I've called it "Ethereum Solidity" throughout the documentation. 

Not true. By just having a very quick look at the docs, I find _many_ instances we just talk about `Solidity`. No mention of `Ethereum Solidity`, `Solang Solidity` or whathever else. Just `Solidity`. And in this case I assume anyone reaading the docs will `Solidity` as to what's documented on docs.soliditylang.org and implemented by solc.

For example in the Solana section. The first sentence:

    As the underlying Solana environment is different than that of Ethereum, Solidity inner workings have been modified to function properly.

So, this should mention that the inner workings of `Ethereum Solidity` have been modified, not `Solidity`. There are _many_ more of these instances.

> If you're going to change this, then change all the links that say "Ethereum Solidity"

As pointed out, in our docs, this is not at all consistent. In fact, there are many instances we write just `Solidity`, without referring to either what you mean with "Ethereum Solidity", nor to "Solang Solidity" nor "Solidity on Solana" nor anything else. So why not do the only sane thing here which is referring the term `Solidity` as to what everyone else in the world refers it to.

Reply:
> > In one hand, Solidity for Solana has its own syntax, so we can see it as a different flavor. Calling Ethereum Solidity to refer to the original Solidity implementation is reasonable, given that we are slightly different than the original.
> 
> Exactly. On Polkadot there are also many small differences with Ethereum Solidity (or should be called Solidity on Ethereum/ Solidity on EVM)?

I'm fine with calling it `Solana Solidity` or `Solidity on Solana` or `Solang Solidity` or whatever. I'm just not convinced that we can not just use `Solidity` when referring to what the World understands as `Solidity` to be is what's documented on soliditylang.org and in `solc`.

And this is _very_ inconsistent in our docs.

But fine. Apparently it's the end of the world if someone talks about `Solidity` in our docs. I'm just gonna `s/Solidity/Ethereum Solidity` here to end this stupid discussion for good

Reply:
Ah, now I get what you mean.

But it doesn't work like that.

You can't invent random German sounding words and start using in them in a local community still call your Language German.

What you end up is a _dialect_.

So the correct thing to say would be that Solang is a Compiler for _Solidity Dialects_ running on other runtimes.

But that still leaves the solc/docs.soliditylang.org the authority. We can't just deny that fact.

Reply:
I am fine with dialect. 

> But that still leaves the solc/docs.soliditylang.org the authority. We can't just deny that fact.

I am totally denying that fact. 

Just in the same way. there is no version German which is not a dialect. Each and every one is a dialect. Just in the same way, the King of England speaks a dialect.

Reply:
I summarize this discussion into the following points:

- Neither the `solc` compiler nor the `docs.soliditylang.org` resemble a language spec.
- Hence it's questionable whether this is authoritative because it's merely the predominant dialect or version of Solidity at the time of writing.
- Add to that, seeing the compiler or docs of a version of Solidity that can only run on a specific runtime is definitively not what we want.
- From that follows, saying just `Solidity` while linking to some specific documentation or implementation should be avoided. Because it implies that the documentation or implementation linked to to be an authoritative source while it is not.

Reply:
@seanyoung we can (we should!) continue discussion in #1479

as for this PR I can just change it to `Ethereum Solidity` because that's what seems to be used most often in other places.